internal/applier/bundleinstance_applier.go

```diff
@@ -1,56 +1,36 @@
 package applier
 
 import (
-	"context"
-
 	rukpakv1alpha1 "github.com/operator-framework/rukpak/api/v1alpha1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 
 	platformv1alpha1 "github.com/openshift/api/platform/v1alpha1"
-	"github.com/openshift/platform-operators/internal/sourcer"
 )
 
 const (
 	plainProvisionerID    = "core-rukpak-io-plain"
 	registryProvisionerID = "core-rukpak-io-registry"
 )
 
-type bdApplier struct {
-	client.Client
-}
-
-func NewBundleDeploymentHandler(c client.Client) Applier {
-	return &bdApplier{
-		Client: c,
-	}
-}
+func NewBundleDeployment(po *platformv1alpha1.PlatformOperator, image string) *rukpakv1alpha1.BundleDeployment {
+	bd := &rukpakv1alpha1.BundleDeployment{}
+	bd.SetName(po.GetName())
 
-func (a *bdApplier) Apply(ctx context.Context, po *platformv1alpha1.PlatformOperator, b *sourcer.Bundle) error {
-	bi := &rukpakv1alpha1.BundleDeployment{}
-	bi.SetName(po.GetName())
 	controllerRef := metav1.NewControllerRef(po, po.GroupVersionKind())
+	bd.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
 
-	_, err := controllerutil.CreateOrUpdate(ctx, a.Client, bi, func() error {
-		bi.SetOwnerReferences([]metav1.OwnerReference{*controllerRef})
-		bi.Spec = *buildBundleDeployment(b.Image)
-		return nil
-	})
-	return err
+	bd.Spec = BuildBundleDeployment(image)
+	return bd
 }
 
-// buildBundleDeployment is responsible for taking a name and image to create an embedded BundleDeployment
-func buildBundleDeployment(image string) *rukpakv1alpha1.BundleDeploymentSpec {
-	return &rukpakv1alpha1.BundleDeploymentSpec{
+// BuildBundleDeployment is responsible for taking a name and image to create an embedded BundleDeployment
+func BuildBundleDeployment(image string) rukpakv1alpha1.BundleDeploymentSpec {
```

Comment:
What's the need to export this?

Reply:
Updated in https://github.com/openshift/platform-operators/pull/41/commits/28b4316bc8c76cdb4d61f41877fbe6351ac50f39.